metricproviders/prometheus/prometheus.go

```diff
@@ -162,14 +165,28 @@ func NewPrometheusAPI(metric v1alpha1.Metric) (v1.API, error) {
 	} else {
 		return nil, errors.New("prometheus address is not configured")
 	}
-	client, err := api.NewClient(api.Config{
-		Address: metric.Provider.Prometheus.Address,
-	})
-	if err != nil {
-		log.Errorf("Error in getting prometheus client: %v", err)
-		return nil, err
+
+	//Check if using Amazon Managed Prometheus if true build sigv4 client
+	promUrl := metric.Provider.Prometheus.Address
+	amznUrlSubstring := "aps-workspaces.region.amazonaws.com"
```

Comment:
Based on there docs I am seeing urls such as `aps-workspaces.${AWS_REGION}.amazonaws.com` so I think what you had before made more sense and just checking for aps-workspaces I was just hoping for something more robust however I am ok with just using `aps-workspaces` if that makes sense for now

Reply:
Just saw some of the urls from here https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-ingest-metrics-existing-Prometheus.html